docs/developers/developer-reference/sc-contract-calls.md

```diff
@@ -252,74 +566,253 @@ fn my_endpoint_callback(&self, #[call_result] result: ManagedAsyncCallResult<Big
 }
 ```
 
-Keep in mind you do NOT need to specify the payments when hooking the callback:
+The differences are:
+- Calling `async_call_promise` instead of `async_call`.
+- Calling `register_promise` instead of `call_and_exit`. This one does not terminate execution.
+- Annotation `#[promises_callback]` instead of `#[callback]`.
+- We need to specify the gas for the call, because the execution of our transaction will continue and it needs to know how much gas it can keep.
+- We need to specify the amount of gas for the callback. This is the exact amount of gas reserved for the callback, irrespective of how much the target contract consumes.
+
+
+
+[comment]: # (mx-context-auto)
+
+### Transfer-execute
+
+Transfer-execute calls are similar to asynchronous calls, but they can have no callback, and thus the caller cannot react in any way to what happens with the callee.
+
+Just like promises, there can be multiple such calls launched from a transaction, but unlike promises, they are alreay available on mainnet.
+
+Transfer-execute calls do not need any further configuration (other than explicit an explicit gas limit), and therefore there is no specific object type associated with them. They can be launched immediately:
+
 ```rust
-#[endpoint]
-fn caller_endpoint(&self) {
-	// other code here
+self.contract_proxy(callee_sc_address)
+	.my_endpoint(my_biguint_arg)
+	.with_gas_limit(gas_limit)
+	.transfer_execute();
+```
 
-	self.contract_proxy(callee_sc_address)
-		.my_endpoint(my_biguint_arg)
-		.async_call()
-		.with_callback(self.callbacks().my_endpoint_callback())
-		.call_and_exit();
-}
+
+
+[comment]: # (mx-context-auto)
+
+### Synchronous calls
+
+Synchronous calls are executed inline: this means execution is interrupted while they are executed, and resumed afterwards. We also get the result of the execution right away and we can use it immediately in the transaction.
```

Comment:
```suggestion
Synchronous calls are executed inline: this means execution is interrupted while they are executed and resumed afterwards. We also get the result of the execution right away and we can use it immediately in the transaction.
```

Reply:
I would add caller and callee to the explaination.
`this means the caller execution is interrupted while the callee function is executed`